### DISCUSSION TITLE
Fetch data by MAX value and subquery

### DISCUSSION THREAD

**Discussion by todordev** [prisma/prisma#6229] [Q&A] [answered]
I would like to fetch data by MAX value of its column.

My SQL query

```sql
SELECT * FROM
    `my_table`
WHERE
    pod_id = 556 
    AND 
    job_id = (
        SELECT
            MAX(job_id)
        FROM
            `my_table`
        WHERE
            pod_id = 556
    )
```

Prisma schema
```prisma
model my_table {
  id           Int           @id @default(autoincrement())
  job          job           @relation(fields: [job_id], references: [id])
  job_id       Int
  pod          pod           @relation(fields: [pod_id], references: [id])
  pod_id       Int
}
```

I am trying to do this, but there is no result. :)

```javascript
let data = await prisma.my_table.findMany({
	where: {
	  pod_id: 556,
	  max: {
	     job_id: true
         }
   }
});
```

What is the best way to get the result by MAX value of a column using Prism?

How to do subqueries via Prisma? 
Are there any documentation about subqueries for Prisma?

Thank you in advance!

**Comment by ryands17** [comment]
@ITPrism 👋 

This is currently not supported by Prisma. I would suggest creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for the same. As a workaround, you can use [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) to achieve this.

**Reply by todordev** [reply]
Thank you very much! :)

**Comment by nmktad** [comment]
@ITPrism 

you can first sort it in descending order. you can use order by and the value after that use `take` to select the amount of results you want to 

```javascript
let data = await prisma.your_table.findMany({
    where: {
	pod_id: 556,
	orderBy: [
          {
            the_value: 'desc'
          }
        ],
        take: 3 or other number of value
     }
});
```

**Reply by MarcoFFSiciliano** [reply]
Thank you

**Comment by Michael-Reich** [comment]
Didn't work for me, but this did.

```
let data = await prisma.your_table.findMany({
  where: {
    pod_id: 556,
  },
  orderBy: [
    {
      the_value: 'desc'
    }
  ],
  take: 3
});
```

**Comment by AtuI-Yadav** [comment, marked as answer]
You can use an aggregate method of the Prisma 
```
let data = await prisma.my_table.aggregate({
	where: {
	  pod_id: 556,
	  }
	  _max: {
	     job_id: true
         }
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
